blpapijs.cpp

```diff
@@ -60,16 +60,38 @@
 #endif
 
 #define BLPAPI_EXCEPTION_TRY try {
-#define BLPAPI_EXCEPTION_CATCH                                              \
-    } catch (blpapi::Exception& e) {                                        \
-        NoRetThrowException(Exception::Error(                               \
-            NEW_STRING(e.description().c_str())));                          \
+
+#define BLPAPI_EXCEPTION_NEW(type)                                          \
+    Local<Object> err = Object::New(Isolate::GetCurrent());                 \
```

Comment:
I really don't want us throwing non-`Error` objects. I think this is a good case for including a module such as [custom-error-generator](https://www.npmjs.com/package/custom-error-generator) and using that to create a specific `Error` type for each SDK exception. It will make a lot more sense when they appear and are logged in JS.